### PR TITLE
Data models

### PR DESCRIPTION
**Pull request by awais-amjed**
In this Pull Request, the following changes were introduced:

- Added Data Models for all the tables in ERD. ✅ 
- Added associations to the Data Models. ✅ 
- Added required functions in the Model Classes. ✅ 
- Added function to get the current user, ✅

### CODE REVIEW

File: app/models/post.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+class Post < ApplicationRecord
+  belongs_to :user
+
+  def increment_counter
+    current_user = User.find(user_id)
```
Comment (micheaol) [inline app/models/post.rb]: Please ensure that your `post model` looks like the code below according to the project requirement:

````
belongs_to :author, class_name: 'User'
has_many :comments
has_many :likes
`````

**Review by micheaol — CHANGES_REQUESTED** [review]
Hi @awais-amjed , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## TO HIGHLIGHT: 
- Your project is professional 💯 
- Your repo is professional 💯 
- Well structured files 💯 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: app/models/post.rb (hunk 1 above)
Comment (awais-amjed) [reply app/models/post.rb]: Ah Thanks, I forgot about that.

**Review by elyor-doniyorov — APPROVED** [review]
Hi @awais-amjed, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
✔️ There are no linter errors
✔️ Descriptive PR title and description

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**